### PR TITLE
Use eksctl when creating the IAM role and trust relationship

### PR DESCRIPTION
**Pull request by ruzickap**
### Description of your changes

Simplify using IRSA by using `eksctl` for creating IAM role and IAM role trust relationship.

### CODE REVIEW

File: AUTHENTICATION.md [hunk 1]
```diff
@@ -110,52 +110,10 @@ eksctl utils associate-iam-oidc-provider --cluster <cluster-name> --region <regi
...
-aws iam create-role --role-name $IAM_ROLE_NAME --assume-role-policy-document file://trust.json --description "IAM role for provider-aws"
-```
-
-Associate a policy with the IAM role. This example uses `AdministratorAccess`,
-but you should select a policy with the minimum permissions required to
-provision your resources.
-
-```
-aws iam attach-role-policy --role-name $IAM_ROLE_NAME --policy-arn=arn:aws:iam::aws:policy/AdministratorAccess
+eksctl create iamserviceaccount --cluster <cluster-name> --region <region> --name="$SERVICE_ACCOUNT_NAME" --namespace="$SERVICE_ACCOUNT_NAMESPACE" --role-name="$IAM_ROLE_NAME" --role-only --attach-policy-arn="arn:aws:iam::aws:policy/AdministratorAccess" --approve
```
Comment (chlunde) [inline AUTHENTICATION.md:116]: Can you use a wildcard for the service account name?

The policy in the current document uses `StringLike`:

```json
        "StringLike": {
          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:provider-aws-*"
        }
```

Comment (ruzickap) [reply AUTHENTICATION.md:116]: Using wildcard is not possible in `eksctl`.

Comment (chlunde) [reply AUTHENTICATION.md:116]: @ruzickap OK, until we can pin the service account name (https://github.com/crossplane/crossplane/issues/2295) - I think we should have a warning here about that.

File: AUTHENTICATION.md [hunk 2]
```diff
@@ -108,7 +108,17 @@ Console](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-s
...
-6. Create IAM Role that provider-aws will use
+6. Create IAM Role that provider-aws will use using `eksctl`
+
+Create IAM role with trust relationship:
+
+```
+eksctl create iamserviceaccount --cluster <cluster-name> --region <region> --name="$SERVICE_ACCOUNT_NAME" --namespace="$SERVICE_ACCOUNT_NAMESPACE" --role-name="$IAM_ROLE_NAME" --role-only --attach-policy-arn="arn:aws:iam::aws:policy/AdministratorAccess" --approve
+```
+
+> The varibale `$SERVICE_ACCOUNT_NAME` contains default service account name and changes with every provider release.
```
[the remark this answers is not in the document]
Comment (ruzickap) [reply AUTHENTICATION.md]: Fixed

File: AUTHENTICATION.md [hunk 3]
```diff
@@ -108,7 +108,17 @@ Console](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-s
 eksctl utils associate-iam-oidc-provider --cluster <cluster-name> --region <region> --approve
 ```
 
-6. Create IAM Role that provider-aws will use
+6. Create IAM Role that provider-aws will use using `eksctl`
+
+Create IAM role with trust relationship:
+
+```
+eksctl create iamserviceaccount --cluster <cluster-name> --region <region> --name="$SERVICE_ACCOUNT_NAME" --namespace="$SERVICE_ACCOUNT_NAMESPACE" --role-name="$IAM_ROLE_NAME" --role-only --attach-policy-arn="arn:aws:iam::aws:policy/AdministratorAccess" --approve
```
[the remark this answers is not in the document]
Comment (ruzickap) [reply AUTHENTICATION.md:116]: Yeah... In some parts of the documentation it is being used `<cluster-name>` or `<region>` and somewhere else `${SERVICE_ACCOUNT_NAMESPACE}`.

I would prefer to use bash variable like `${CLUSTER_NAME}` everywhere if you agree...?

**Review by chlunde — APPROVED** [review]

File: AUTHENTICATION.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ruzickap) [reply AUTHENTICATION.md]: Fixed...

**Comment by chlunde** [comment]
@ruzickap looks good, I see you did some final touches, now even `shellcheck` would approve :smile:  If you're done I think we can merge. OK?

**Comment by ruzickap** [comment]
Sure... Go ahead...

**Pull request merged**